packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -1136,7 +1142,35 @@ abstract class State<T extends StatefulWidget> with Diagnosticable {
   ///    from the tree permanently.
   @protected
   @mustCallSuper
-  void deactivate() { }
+  void deactivate() {
+    assert(() {
+      _debugActive = !_debugActive;
+      return _debugActive == false;
+    }());
+  }
+
+  /// Called when this object is reactivated.
+  ///
+  /// If the [widget] or one of its ancestors has a [GlobalKey], the framework
+  /// will mark this object as inactive when it is removed and call
+  /// [deactivate].
+  ///
+  /// If the object is reinserted to the tree in the next frame (e.g. by
+  /// changing position), it will be marked as active again and this method will be
+  /// called.
+  ///
+  /// See also:
+  ///
+  ///  * [Element.activate] and [Element.deactivate] for more information about
+  ///  lifecycle.
+  @protected
+  @mustCallSuper
+  void reactivate() {
+    assert(() {
+      _debugActive = !_debugActive;
+      return _debugActive == true;
```

Comment:
`== true` is redundant here. I'm guessing that's one of the things the analyzer shard is complaining about, though I haven't checked.